File: app/src/at-myhuynh/java/asiantech/internship/summer/java_core/exoop/Rectangle.java
package asiantech.internship.summer.java_core.exoop;

public class Rectangle extends Shape {

    private int length;
    private int width;

    public Rectangle(int length, int width) {
        this.length = length;
        this.width = width;
    }

    @Override
    public double perimeter() {
        return (length + width) * 2;
    }

    @Override
    public double area() {
        return length * width;
    }

    @Override
    public String print() {
        if (length == width) {
            return "HV";
        } else {
            return "HCN";
        }
    }
}

File: app/src/at-myhuynh/java/asiantech/internship/summer/kotlin/exoop/Rectangle.kt
package asiantech.internship.summer.kotlin.exoop

class Rectangle(val length: Int, val width: Int) : Shape() {

    override fun perimeter(): Double {
        return ((length + width) * 2).toDouble()
    }

    override fun area(): Double {
        return (length * width).toDouble()
    }

    override fun print(): String {
        return if (length == width) {
            "HV"
        } else {
            "HCN"
        }
    }
}
